### ISSUE TITLE
Wrong error location when missing a bound on a Copy/Clone impl for a generic struct containing a field that has an associated type as type.

### ISSUE THREAD

**Issue by jix** [rust-lang/rust#56064]
When compiling this code, which misses `T: SomeTrait` bounds in the impls, the missing bounds are reported for the struct, which does already have that bound:

[playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=b2da3793b310639f5441d317c9302626)
```rust
trait SomeTrait {
    type SomeAssociatedType: Copy;
}

// Reported error location is in the line below
struct SomeStruct<T: SomeTrait>(T::SomeAssociatedType);

// The actual errors are missing T: SomeTrait bounds below
impl<T> Copy for SomeStruct<T> {}

impl<T> Clone for SomeStruct<T> {
    fn clone(&self) -> Self {
        *self
    }
}
```

The error message is especially confusing as it suggests adding an already existing bound and offers no help in locating the actual issue:
```
error[E0277]: the trait bound `T: SomeTrait` is not satisfied
 --> src/lib.rs:5:33
  |
5 | struct SomeStruct<T: SomeTrait>(T::SomeAssociatedType);
  |                                 ^^^^^^^^^^^^^^^^^^^^^ the trait `SomeTrait` is not implemented for `T`
  |
  = help: consider adding a `where T: SomeTrait` bound
```

When implementing an empty trait instead of `Copy` and `Clone` the error is reported for the correct location.

This is manually reduced as much as I could. In a larger code base it is much harder to figure out the location of the missing trait bound.

I tested on playground with stable/nightly and 2015/2018 as well as locally with
```
rustc 1.32.0-nightly (13c943992 2018-11-18)
binary: rustc
commit-hash: 13c9439925797cd7a65c917d047c07a500d9bfe6
commit-date: 2018-11-18
host: x86_64-unknown-linux-gnu
release: 1.32.0-nightly
LLVM version: 8.0
```

**Comment by estebank** [comment]
Triage: we no longer suggest the incorrect code, otherwise no change.

**Comment by estebank** [comment]
The current output takes you to working code, in stages:

```
error[[E0277]](https://doc.rust-lang.org/nightly/error-index.html#E0277): the trait bound `T: SomeTrait` is not satisfied
 --> src/lib.rs:8:18
  |
8 | impl<T> Copy for SomeStruct<T> {}
  |                  ^^^^^^^^^^^^^ the trait `SomeTrait` is not implemented for `T`
  |
help: consider restricting type parameter `T`
  |
8 | impl<T: SomeTrait> Copy for SomeStruct<T> {}
  |       +++++++++++
```
```
error[[E0277]](https://doc.rust-lang.org/nightly/error-index.html#E0277): the trait bound `T: SomeTrait` is not satisfied
 --> src/lib.rs:8:18
  |
8 | impl<T> Copy for SomeStruct<T> {}
  |                  ^^^^^^^^^^^^^ the trait `SomeTrait` is not implemented for `T`
  |
help: consider restricting type parameter `T`
  |
8 | impl<T: SomeTrait> Copy for SomeStruct<T> {}
  |       +++++++++++
```
